docs/generating-code-from-schema.md

```diff
@@ -230,30 +230,140 @@ Finally, `maxProjectionDepth` will instruct codegen to stop generating beyond 2
 The default is 10.
 This will help further limit the number of projections as well.
 
+### Generating classes with Custom Annotations
+This feature provides the ability to support any custom annotation on the generated POJOs using the @annotate directive in graphQL.
+The @annotate directive can be placed on type, input or fields in the graphQL. This feature is turned off by default and can be enabled by setting generateCustomAnnotation to true in build.gradle.
+```groovy
+generateJava {
+    ...
+    generateCustomAnnotations = true
+}
+```
+@annotate contains 3 fields:
+* name - Mandatory field. Name of the annotation. Eg: ValidPerson. You can have the package along with the annotation name. eg: com.test.ValidPerson. The package value given with the annotation name takes precedence over the mapped package in build.gradle.
+* type - Optional field. This variable is used to map the annotation package in build.gradle. The package if given with annotation name will take precedence over this value. But if neither are given an empty string is used.
+* inputs - Optional field. Contains the inputs to the annotation in key-value pairs. Eg: inputs: {types: [HUSBAND, WIFE]}
+
+@annotate definition in the graphQL:
+```
+"Custom Annotation"
+directive @annotate(
+    name: String!
+    type: String
+    inputs: JSON
+) repeatable on OBJECT | FIELD_DEFINITION | INPUT_OBJECT | INPUT_FIELD_DEFINITION
+```
+Make sure that the corresponding Java annotations corresponding to the annotation names in the schema should be implemented by the user in their project.
```

Comment:
Please rephrase `Make sure that the corresponding Java annotations corresponding to the annotation names in the schema should be implemented by the user in their project` to `Custom annotations specified in the schema will require corresponding Java implementations to be implemented by the user in their project`